### ISSUE TITLE
Cannot use named volumes for `/var/www/app/public`

### ISSUE THREAD

**Issue by zaxbux** [invoiceninja/dockerfiles#373]
**Setup information**
docker-compose

**Describe the bug**
The current Dockerfile for v5 does not create the `/var/www/app/public` directory, preventing use of named volumes at runtime.

See: https://github.com/docker/compose/issues/3270#issuecomment-543603959

**To reproduce**

docker-compose.yaml
```yaml
volumes:
  invoiceninja_storage:
  invoiceninja_public:

services:
  server:
    image: nginx
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf:ro
      - "invoiceninja_public:/var/www/app/public:ro"

  app:
    image: invoiceninja/invoiceninja:5
    volumes:
      - "invoiceninja_public:/var/www/app/public:rw,delegated"
      - "invoiceninja_storage:/var/www/app/storage:rw,delegated"
    depends_on:
      - db

# db service, etc...
```

**Expected behavior**
The `/var/www/app/public` directory should be created during build, e.g.:

```dockerfile
...

RUN mkdir /var/www/app/public && chown -R $UID:$UID /var/www/app/public

USER $UID

...
```

Currently, it looks like that directory is moved during build and then re-created during runtime.

**Screenshots/logs**

On `docker-compose up`:

```
Recreating invoiceninja-v5_1 ... done
Attaching to invoiceninja-v5_1
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/assets': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/css': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/favicon.ico': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/favicon.png': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/flutter_service_worker.js': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/icons': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/images': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/index.php': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/js': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/logo': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/main.dart.js': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/main.dart.js.map': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/main.foss.dart.js': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/main.foss.dart.js.map': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/main.wasm.dart.js': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/manifest.json': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/mix-manifest.json': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/robots.txt': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/svg': Permission denied
invoiceninja-v5_1        | cp: can't create directory '/var/www/app/public/vendor': Permission denied
invoiceninja-v5_1        | cp: can't create '/var/www/app/public/version.json': Permission denied
```

**Docker**:
- Output of `docker version`:
```
Client: Docker Engine - Community
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:38 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:50 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Additional context**

**Comment by lwj5** [comment]
Hi @zaxbux 

this is a permission issue. Have you tried following the readme and chown the folders 1500

**Comment by zaxbux** [comment]
That works for host volumes, however I'm specifically using named volumes. That would require me to execute commands inside the container, but it keeps exiting due to the init script not being able to create the directory and set permissions.

**Comment by lwj5** [comment]
you can execute the commands using another container. See option 2: https://serverfault.com/questions/984578/change-permissions-for-named-volumes-in-docker You will need it for storage volume as well I presume. 

For option 1, Would you like to create a PR?

**Comment by LaurensRietveld** [comment]
For future reference, an alternative is running these two commands before trying to start invoiceninja:

```
docker-compose run --rm --user root --entrypoint "/bin/sh -c"  app "chmod 755 /var/www/app/public"
docker-compose run --rm --user root --entrypoint "/bin/sh -c" app "chown -R 1500:1500 /var/www/app"
```

**Comment by DLeich** [comment]
Just stumbled into this problem while I was trialing Invoice Ninja on my own utilizing Docker Compose, but attempting from purely a git-ops perspective.  If you want the provisioning 100% automated, then the manual steps are not acceptable.  I was able to add a configuration step using a separate container in my docker-compose.yml.  Inject this in your services configuration of your compose file:

```
  setup:
    image: ubuntu:22.04
    entrypoint:
      - bash
      - -c
    command:
      - >
        chmod 755 /var/www/app/public;
        chown -R 1500:1500 /var/www/app;
    volumes:
      - public:/var/www/app/public:rw,delegated
      - storage:/var/www/app/storage:rw,delegated
```

Ubuntu docker container is probably way over inflated, but I just grabbed it for ease in this case.  This did work for me though so now I have a 100% automated deployment with zero manual steps.

You'll note that my volumes are named and persistent, as opposed to the recommended steps that utilizes bind mounts to what you would clone down from git as documented in the readme.  Technically possible from the command line for tinkering around, but I would consider that to not be best practice for Docker Compose, for the exact reason this issue was created.
